demisto_sdk/commands/format/update_generic.py

```diff
@@ -220,98 +219,80 @@ def regex_matching_key(self, field, schema_keys):
         else:
-            if LooseVersion(self.data.get(self.from_version_key, DEFAULT_CONTENT_ITEM_FROM_VERSION)) < \
-                    LooseVersion(GENERIC_OBJECTS_DEFAULT_FROMVERSION):
-                self.data[self.from_version_key] = GENERIC_OBJECTS_DEFAULT_FROMVERSION
+            click.secho('Skipping update of fromVersion', fg='yellow')
+            return False
 
-    def set_fromVersion(self, from_version=None, file_type: Optional[str] = None):
-        """Sets fromVersion key in file:
+    def should_update_fromversion(self, max_version, current_fromversion_value, file_type):
+        """
+        Check if should change the fromversion key.
         Args:
-            from_version: The specific from_version value.
-            file_type: what is the file type: for now only integration type passed
+            max_version: the maximum version between general, default and current.
+            current_fromversion_value: current from_version if exists in the file.
+            file_type: the file type.
+        Returns:
+            int.
+            1 if its an old file type.
+            0 if shouldn't change the fromversion.
+            -1  if the max version different from the maximum fromversion.
         """
-        metadata = get_pack_metadata(self.source_file)
-        # if it is new contributed pack = setting version to 6.0.0
-        should_set_from_version = ((metadata.get('currentVersion', '') == '1.0.0') and
-                                   (metadata.get('support', '') != 'xsoar'))
+        if file_type and file_type in OLD_FILE_TYPES:
+            return 1
+        if max_version != current_fromversion_value:
+            return -1
+        return 0
 
-        # If file type is a generic object (generic field/type/module/definition) - fromVersion should be at least 6.5.0
-        if file_type in GENERIC_OBJECTS_FILE_TYPES:
-            self.set_fromVersion_of_generic_object(from_version)
+    def set_default_from_version(self, default_from_version: str, current_fromversion_value: str, file_type: str):
+        """
+        Sets the default fromVersion key in the file:
+            In case the user approved it:
+                Set the fromversion to 5.0.0 for old content items.
+                Set/update the fromversion to the input default if supplied.(checks if it is the highest one).
+                In any other case set it to the general one.
+        Args:
+            default_from_version: default fromVersion specific to the content type.
+            current_fromversion_value: current from_version if exists in the file.
+            file_type: the file type.
+        """
+        max_version = get_max_version([GENERAL_DEFAULT_FROMVERSION, default_from_version, current_fromversion_value])
+        should_update = self.should_update_fromversion(max_version, current_fromversion_value, file_type)
+        if should_update and (self.assume_yes or self.ask_user()):
+            if should_update == 1:
```

Comment:
overall this is ok and works as should, sharing my thought anyway:
IMO there is no need in the new function (which returns integer values while its function implies it should return a boolean). I'd simply do something like:
```suggestion
        if max_version != current_fromversion_value and (self.assume_yes or self.ask_user()):
            if file_type and file_type in OLD_FILE_TYPES:
```
note that it's one line less, one less function call and a bit more straight-forward.